common/types/mapper/thrift/history.go

```diff
@@ -217,9 +217,9 @@ func ToGetMutableStateResponse(t *history.GetMutableStateResponse) *types.GetMut
 	return &types.GetMutableStateResponse{
 		Execution:                            ToWorkflowExecution(t.Execution),
 		WorkflowType:                         ToWorkflowType(t.WorkflowType),
-		NextEventID:                          t.NextEventId,
+		NextEventID:                          t.GetNextEventId(),
 		PreviousStartedEventID:               t.PreviousStartedEventId,
-		LastFirstEventID:                     t.LastFirstEventId,
+		LastFirstEventID:                     t.GetLastFirstEventId(),
```

Comment:
not specific to this one, and I know it's sorta late in these reviews to add this, but: tbh I'm kinda getting increasingly concerned about not detecting "things we think are always sent, but occasionally are not".

is it worth making a `notnull(thing)` helper to panic if that expectation turns out wrong?  presumably we have a per-request `defer func() { recover() }()` in some form (e.g. rpc layer) so it just becomes an error, so it'd be "safe" to do so.  it'd probably only be needed / most-valuable in these mapper files.

Reply:
We actually are not changing behavior when the field is not sent occasionally. Currently pretty much everywhere pointer fields are retrieved with `.GetThing()` helper. Thus it may be the case that those fields are not sent, but code consuming it is OK with zero value that `.GetThing()` helper returns. So I'm essentially moving where this helper is called. Instead of being scattered around the codebase it will immediately be called in RPC mapping layer. Having `notnull(thing)`  helper that panics would actually change this behavior.

Of course, there are several places that do not use `GetThing` helpers. They check for nil values and do something about that. I'm leaving such fields out of these PRs, and will address them separately.

Reply:
so that'd be the rare exception like... domain was it?  which had a bigger branch on null than just the internal "get" helpers.

yea, that makes sense - they'd be rather visible during review I guess.  and no doubt there are a lot of these "nobody checks for nil" fields.